tensorboard/plugins/npmi/add_embeddings.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+from absl import app
+from absl import flags
+from tensorboard.plugins.npmi import summary
+from tensorboard.plugins.npmi import metadata
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    "log_path", "", "Log file to extend by embedding data.",
+)
+flags.DEFINE_string(
+    "out_path", "", "Where to write the new log file.",
+)
+flags.DEFINE_string(
+    "embeddings_path", "", "Where to write the new log file.",
+)
+
+
+def add_embeddings(log_path, out_path, embeddings_path):
+    with open(embeddings_path, "rb") as f:
```

Comment:
Does this work if all log_path, out_path, and embeddings_path are empty strings? Shouldn't there be some protection against that?

Reply:
Changed flags to be required.